extra/string-extra/src/StringExtension.php

```diff
@@ -21,11 +23,17 @@ public function getFilters()
     {
         return [
             new TwigFilter('u', [$this, 'createUnicodeString']),
+            new TwigFilter('slug', [$this, 'createSlug']),
         ];
     }
 
     public function createUnicodeString(?string $text): UnicodeString
     {
         return new UnicodeString($text ?? '');
     }
+
+    public function createSlug(string $string, string $separator = '-', string $locale = null): AbstractUnicodeString
```

Comment:
making the locale optional here seems weird to me. The behavior in AsciiSlugger means that it keeps the current locale set in the slugger. But as you instantiate the slugger here rather than using DI, there is no way to control the locale of the slugger.

Reply:
What do you propose @stof ?

Reply:
IMO, the Slugger should be injected in the extension with DI. This allows this instance to be integrated with your locale management to benefit from it being locale aware.